src/tiledb/cloud/geospatial/helpers.py

```diff
@@ -0,0 +1,188 @@
+                        maxy=src.bounds[3]
+                    )
+
+                    # initialization of variables
+                    if proj is None:
+                        proj = src.crs
+                    else:
+                        if proj != src.crs:
+                            raise ValueError("All datasets need to be in the same projection")
+                    
+                    if num_channels is None:
+                        num_channels = src.count
+                    else:
+                        if num_channels != src.count:
+                            raise ValueError("All datasets need to have the same number of bands")
+
+                    if xres is None:
+                        xres, yres = src.res
+
+            if not extents:
+                continue
+
+            if full_extents:
+                full_extents = GeoExtents(*shapely.unary_union([shapely.box(*full_extents.bounds), shapely.box(*extents.bounds)]).bounds)
+            else:
+                full_extents = extents
+ 
+            yield GeoMetadata(path=pth, extents=extents, projection=proj, block_shapes=(dst_tile_size, dst_tile_size))
+
+    if len(source) == 1 and vfs.is_dir(source[0]):
+        with tiledb.scope_ctx(ctx_or_config=config):
+            # Folder like input
+            contents = vfs.ls(source[0])
+            if len(contents) != 0:
+                return group_by_raster_block(tuple(iter_paths(contents))), full_extents, (xres, yres) 
+            else: 
+                raise ValueError("Input bucket should contain geospatial datasets for ingestion")
+    elif isinstance(source, Sequence):
+        if len(source) > 0:
+            # List of input uris - single file is one element list
+            return group_by_raster_block(tuple(iter_paths(source))), full_extents, (xres, yres)
+        else:
+            raise ValueError("Input sequence should contain geospatial datasets for ingestion")
+
+def serialize_filter(filter):
+    if isinstance(filter, tiledb.Filter):
+        filter_dict = filter._attrs_()
+        filter_dict["_name"] = type(filter).__name__
+        return filter_dict
+    else:
+        raise TypeError
+
+
+def batch(iterable, chunks):
+    # Iterator for providing batches of chunks
+    length = len(iterable)
+    for ndx in range(0, length, chunks):
+        yield iterable[ndx : min(ndx + chunks, length)]
+
+
+def scale_calc(samples, num_batches):
+    """Calculate scaling settings for batch_size and max_workers
+
+    :param source: The source iterable containing files to be ingested/exported
+    :param num_batches: The number of batches given by the API
+    :return: Tuple batch_size, max_workers
+    """
+    # If num_batches is default create number of images nodes
+    # constraint node max_workers to 20 fully heuristic
+    batch_size = 1 if num_batches is None else math.ceil(len(samples) / num_batches)
+    max_workers = 20 if num_batches is None else None
+    return batch_size, max_workers
```

Comment:
This was copied over as a pattern from the biomedical ingestion, but I don't mind moving them into the same module. Having a shared helpers class for calculating batch sizes would be useful.

Reply:
How about src/tiledb/cloud/batching.py if it will be used in >2-3 places? I'm generally 👎 on helpers.py or utils.py because then they tend to become disorganized and hard to maintain.